### ISSUE TITLE
typescript casting issues

### ISSUE THREAD

**Issue by dawidzq** [rt2zz/redux-persist#493]
Hey,

So I'm currently trying to implement redux-persist using angular and typescript. I implemented my module very closely to this person: https://gist.github.com/jorgeramirez/bc30322be89874370e7571b0475714a5. 

Since I'm using the ng-redux angular wrapper, I'm doing something like 
```
export const ExampleModule = module('some-module', [ ngReduxUiRouter, ngRedux ])
.config(($ngReduxProvider: INgReduxProvider) => {
    let middleware = [ /* ... */ ];
    let enhancers = [autoRehydrate()];

    $ngReduxProvider.createStoreWith(reducer, middleware, enhancers);
})
.run(($ngRedux: INgRedux) => {
    persistStore($ngRedux);
})
.name;
```

Everything seems to be working fine, but typescript really doesn't like the `persistStore($ngRedux)` line. The specific error is: 
> Error:(67, 22) TS2352:Type 'INgRedux' cannot be converted to type 'Store<any>'.
>   Types of property 'subscribe' are incompatible.
>     Type '(listener: Function) => Function' is not comparable to type '(listener: () => void) => Unsubscribe'.
>       Type 'Function' is not comparable to type 'Unsubscribe'.
>         Type 'Function' provides no match for the signature '(): void'.

The listener functions are awfully similar, and I'm pretty sure that Unsubscribe simply extends off of Function. Has anyone had this come up? If so, are there any workarounds? I'm still pretty new to redux, so user error isn't off the table.

**Comment by rt2zz** [comment]
Right, so it looks like we just have a bad proxy for the Store type: https://github.com/rt2zz/redux-persist/blob/v5/src/types.js#L80

Maybe we just loosen that up and make it return any

**Comment by rt2zz** [comment]
sorry I linked to the flow type, but I think there is something similar on the v4 typescript type

**Comment by dawidzq** [comment]
I'm wondering if the issue could lie in ng-redux, actually. ng-redux declares the return type of "subscribe" as `Function` when `Unsubscribe` seems like it would be more appropriate. I opened a PR here: https://github.com/angular-redux/ng-redux/pull/173.

I wouldn't be able to do this in production, but changing the return type of subscribe to `Unsubscribe` did not break anything and also resolved the Typescript issue mentioned above.

**Comment by rt2zz** [comment]
👍  lmk, I am not using typescript myself, but am committed to keeping up to date definitions in the lib.
